docs/pages/extensibility/ui.md

```diff
@@ -39,8 +39,94 @@ Meshery leverages remote providers for identity management. These providers can
 
 The user avatar behavior, which changes based on the user's status, can be customized by extending the [User Component](https://github.com/meshery/meshery/blob/7de49ef4928f114080f923f2ad261f4433ca91d6/ui/components/User.js#L46).
 
+### Extensibility: Permissions (CASL)
 
-###  Build-Time UI Extensibility
+Meshery UI uses [CASL.js](https://casl.js.org/v6/en/) to implement it's permissions framework. CASL is an isomorphic authorization JavaScript library which restricts what resources a given client is allowed to access.
+It has been designed to be very customizable and robust.
+It uses keys as it's unit checks to see if a permission to perform or action or not.
+
+#### Implementation
+
+Once a user has logged in, the backend will send a response with the permissions that the user has. Those permissions will be used to check abilities on the frontend.
```

Comment:
We should be mentioning that these keys are grouped under keychains and they are assigned to roles then those roles are assigned to users. Then the server checks what roles does user has and they are access to do.  Not directly backend will send a response.

Reply:
@sudhanshutech this is great info, however, that functionality is not offered in Meshery directly. Providers can extend Meshery in the way in which you described, if they like, though.